### PR TITLE
OFVChallenger Part 1

### PR DESCRIPTION
**Pull request by kongzii**
This is Part 1 of two part series for OFVChallengerAgent (to keep PRs smaller).

This more or less takes the PR from Valory's mech and implements it here, but refactored for our nicer codebase:

- Use APIKeys where possible
- Strong typing
- Langfuse support
- Use functions from PMAT

### CODE REVIEW

**Comment by kongzii** [comment]
Here is TSV file with markets that I manually resolved back then, I'm waiting for the benchmark to finish to verify all is correct.

[markets.tsv.zip](https://github.com/user-attachments/files/16814984/markets.tsv.zip)

**Comment by kongzii** [comment]
Benchmark result:

```
Current accuracy: 73.47%
OFV's accuracy: 89.80%
```

File: prediction_market_agent/agents/ofvchallenger_agent/deploy.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        if market_type != MarketType.OMEN:
+            raise RuntimeError("Can challenge only Omen.")
+
+        self.challenge()
+
+    @observe()
+    def challenge(self) -> None:
+        self.langfuse_update_current_trace(tags=[OFV_CHALLENGER_TAG])
+        APIKeys()
+        # TODO: Coming in the next PR.
```
Comment (evangriffiths) [inline prediction_market_agent/agents/ofvchallenger_agent/deploy.py:21]: 👁️ 👁️

File: prediction_market_agent/utils.py [hunk 2]
```diff
@@ -22,7 +22,9 @@ class DBKeys(BaseSettings):
 
 
 class APIKeys(APIKeysBase):
+    # Don't get fooled! Serper and Serp are two different services.
```
Comment (evangriffiths) [inline prediction_market_agent/utils.py:25]: 👍

File: pyproject.toml [hunk 3]
```diff
@@ -53,6 +53,7 @@ pinatapy-vourhey = "^0.2.0"
 plotly = "^5.22.0"
 prediction-prophet = { git = "https://github.com/agentcoinorg/predictionprophet.git", rev = "69592d0122bff1271d3e5cd6efdf4e1ef240bc06" }
 transformers = "^4.43.3"
+openfactverification-kongzii = "^0.1.0"
```
Comment (evangriffiths) [inline pyproject.toml:56]: what's the reason again that we don't use the [original](https://github.com/Libr-AI/OpenFactVerification)?

**Review by evangriffiths — APPROVED** [review]

File: pyproject.toml (hunk 3 above)
Comment (kongzii) [reply pyproject.toml:56]: They don't have a pypi package and they didn't want to merge my change to include Serper's result date in the prompts, which helped with accuracy in our PMs.

**Pull request merged**